src/Cubix/Language/Parametric/Semantics/Cfg/Graph.hs

```diff
@@ -245,15 +245,19 @@ satisfyingBoundary seen succ pred cfg node =
     if labs == [] then
       lift Nothing
     else do
-      nextLab <- ListT (Just labs)
+      nextLab <- select labs  -- `select ( labs :: [Label] )` produces a `ListT Maybe Label`
       satisfyingBoundary (Set.insert (node ^. cfg_node_lab) seen) succ pred cfg (lookupCfg cfg nextLab)
 
+-- The function `runListT` defined in the `list-transformer` package has the wrong type signature. Let's define a shim.
```

Comment:
How did it compile before?

Reply:
The old `mtl` had the old and not quite correct `ListT` transformer (before the upgrade to latest stack). You might remember that `ListT m`  was initially (in 2001) implemented to work correctly only with commutative monads `m`. By now, both `transformers` and `mtl` have completely removed `ListT` and it now exists only in the new package `list-transformer`. 

I had two choices: copy the old `ListT` from `mtl` before it was deleted, or uptake the new `ListT` that works with all monads.

Reply:
Another possibility is to refactor the code that uses `ListT` if all it really does is some simple manipulations with the data types `Maybe [ a ]` and `[ Maybe a ]`. I haven't yet understood in detail what that code does.

Reply:
> Yeah, I think that's better. Using ListT is not great.

I'm almost done eliminating all usages of `ListT` here.